server/services/token/token_service.go

```diff
@@ -4,40 +4,29 @@ import (
 	"fmt"
 	"main/server/db"
 	"main/server/model"
-	"main/server/response"
-	"main/server/utils"
 	"os"
 	"time"
 
-	"github.com/gin-gonic/gin"
 	"github.com/golang-jwt/jwt/v4"
 )
 
-type Claims struct {
-	UserId string `json:"userId"`
-	Phone  string `json:"phone"`
-	Role   string `json:"role"`
-	jwt.RegisteredClaims
-}
-
 // Generate JWT Token
-func GenerateToken(claims Claims, ctx *gin.Context) *string {
+func GenerateToken(claims model.Claims) (*string, error) {
```

Comment:
Generating token can also be considered as a part of the authentication process.